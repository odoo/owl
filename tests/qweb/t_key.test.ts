describe.skip("t-key", () => {
  test("can use t-key directive on a node", () => {
    // qweb.addTemplate("test", `<div t-key="beer.id"><t t-esc="beer.name"/></div>`);
    // expect(renderToString(qweb, "test", { beer: { id: 12, name: "Chimay Rouge" } })).toBe(
    //   "<div>Chimay Rouge</div>"
    // );
  });

  test("t-key directive in a list", () => {
    // qweb.addTemplate(
    //   "test",
    //   `<ul>
    //     <li t-foreach="beers" t-as="beer" t-key="beer.id"><t t-esc="beer.name"/></li>
    //    </ul>`
    // );
    // expect(
    //   renderToString(qweb, "test", {
    //     beers: [{ id: 12, name: "Chimay Rouge" }],
    //   })
    // ).toMatchSnapshot();
  });
});
