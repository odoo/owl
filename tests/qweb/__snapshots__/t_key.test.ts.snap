// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`t-key can use t-key directive on a node 1`] = `
"function anonymous(bdom, helpers
) {
  let { text, createBlock, list, multi, html, toggler, component } = bdom;
  let { withDefault, getTemplate, prepareList, withKey, zero, call, callSlot, capture, isBoundary, shallowEqual, setContextValue, toNumber } = helpers;
  
  let block1 = createBlock(\`<div><block-text-0/></div>\`);
  
  return function template(ctx, node, key = \\"\\") {
    let d1 = ctx['beer'].name;
    return block1([d1]);
  }
}"
`;

exports[`t-key t-key directive in a list 1`] = `
"function anonymous(bdom, helpers
) {
  let { text, createBlock, list, multi, html, toggler, component } = bdom;
  let { withDefault, getTemplate, prepareList, withKey, zero, call, callSlot, capture, isBoundary, shallowEqual, setContextValue, toNumber } = helpers;
  
  let block1 = createBlock(\`<ul><block-child-0/></ul>\`);
  let block3 = createBlock(\`<li><block-text-0/></li>\`);
  
  return function template(ctx, node, key = \\"\\") {
    ctx = Object.create(ctx);
    const [k_block2, v_block2, l_block2, c_block2] = prepareList(ctx['beers']);
    for (let i1 = 0; i1 < l_block2; i1++) {
      ctx[\`beer\`] = v_block2[i1];
      let key1 = ctx['beer'].id;
      let d1 = ctx['beer'].name;
      c_block2[i1] = withKey(block3([d1]), key1);
    }
    let b2 = list(c_block2);
    return block1([], [b2]);
  }
}"
`;
